client/homepage/stats-overview/test/index.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * External dependencies
+ */
+import { render, fireEvent, screen } from '@testing-library/react';
+import { StatsOverview } from '../index';
+import { recordEvent } from 'lib/tracks';
+
+jest.mock( 'lib/tracks' );
+
+describe( 'StatsOverview tracking', () => {
+	it( 'should record an event when a stat is toggled', () => {
+		render(
+			<StatsOverview
+				userPrefs={ {
+					hiddenStats: null,
+				} }
+				updateCurrentUserData={ () => {} }
+			/>
+		);
+
+		const ellipsisBtn = screen.getByTitle(
+			'Choose which values to display'
+		);
+		fireEvent.click( ellipsisBtn );
+		const totalSalesBtn = screen.getByText( 'Total Sales' );
+		fireEvent.click( totalSalesBtn );
+
+		expect( recordEvent ).toHaveBeenCalledWith(
+			'statsoverview_indicators_toggle',
+			{
+				indicator_name: 'revenue/total_sales',
+				status: 'off',
+			}
+		);
+	} );
+} );
+
+describe( 'StatsOverview stat selection', () => {
```

Comment:
I think this test name could be a bit more descriptive - maybe "StatsOverview toggle and persist stat preference" or something like that.